src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt

```diff
@@ -104,16 +119,40 @@ class VisitNotificationEventService(
     return savedVisitNotificationEvent.reference
   }
 
-  private fun getOverLappingVisits(nonAssociationChangedNotification: NonAssociationChangedNotificationDto): List<VisitDto> {
+  private fun getOverLappingVisits(notificationDto: NonAssociationChangedNotificationDto): List<VisitDto> {
     // get the prisoners' prison code
-    val prisonCode = prisonerService.getPrisoner(nonAssociationChangedNotification.prisonerNumber)?.prisonCode
+    val prisonCode = getPrisonCode(notificationDto.prisonerNumber)
     val fromDate = LocalDate.now().atStartOfDay()
 
-    val primaryPrisonerVisits = visitService.getBookedVisits(nonAssociationChangedNotification.prisonerNumber, prisonCode, fromDate)
-    val nonAssociationPrisonerVisits = visitService.getBookedVisits(nonAssociationChangedNotification.nonAssociationPrisonerNumber, prisonCode, fromDate)
+    val primaryPrisonerVisits = visitService.getBookedVisits(notificationDto.prisonerNumber, prisonCode, fromDate)
+    val nonAssociationPrisonerVisits = visitService.getBookedVisits(notificationDto.nonAssociationPrisonerNumber, prisonCode, fromDate)
     return getOverLappingVisits(primaryPrisonerVisits, nonAssociationPrisonerVisits)
   }
 
+  private fun getAffectedNotifications(notificationDto: NonAssociationChangedNotificationDto): List<VisitNotificationEvent> {
+    // get the prisoners' prison code
+    val prisonCode = getPrisonCode(notificationDto.prisonerNumber)
+
+    val prisonersNotifications = visitNotificationEventRepository.getEventsBy(notificationDto.prisonerNumber, prisonCode, NON_ASSOCIATION_EVENT)
+    val nonAssociationPrisonersNotifications = visitNotificationEventRepository.getEventsBy(notificationDto.nonAssociationPrisonerNumber, prisonCode, NON_ASSOCIATION_EVENT)
+
+    val prisonerEventsToBeDeleted = prisonersNotifications.filter {
+        prisonersNotify ->
+      nonAssociationPrisonersNotifications.any { it.reference == prisonersNotify.reference }
+    }
+    val nsPrisonerEventsToBeDeleted = nonAssociationPrisonersNotifications.filter {
+        nsPrisonersNotify ->
+      prisonerEventsToBeDeleted.any { it.reference == nsPrisonersNotify.reference }
+    }
+
+    return (prisonerEventsToBeDeleted + nsPrisonerEventsToBeDeleted).sortedBy { it.reference }
+  }
+
+  private fun getPrisonCode(prisonerNumber: String): String {
+    return prisonerService.getPrisonerPrisonCode(prisonerNumber)
+      ?: throw ItemNotFoundException("Prison can not be found for prisoner: $prisonerNumber")
```

Comment:
If we get a notification for a non supported prisoner will ItemNotFoundException be thrown? Will the message end up in DLQ as a result?

Reply:
depends how we handling it orch service, not the responsibility of this code, but good point not will look into the orch side of things.

Reply:
may be return another code?